src/sql-parser/src/parser.rs

```diff
@@ -1932,6 +1932,15 @@ impl<'a> Parser<'a> {
             None
         };
 
+        // TODO(chae/aljoscha): is this the right place to generate a shard id? Should it be in normalization?
```

Comment:
I think in the good version it should be the storage controller (maybe?) that knows (and comes up with) the shard IDs for both the data collection and timestamp collection.

And storage needs to remember those shard IDs, but there is currently no state that the storage controller/STORAGE keeps and restores when restarting.

Reply:
makes sense -- guess that'll be a follow-up because there's no mechanism at the moment for storage.

Issue right now with creating it here is that the `sql-parser/tests/testdata/ddl` tests try to hardcode a shard ID but this generates a new one for each invocation so it makes the tests not replicable

Reply:
I've rebased and moved shard generation to `create_sources`